statsmodels/tsa/arima/params.py

```diff
@@ -394,3 +698,88 @@ def __repr__(self):
         if not self.spec.concentrate_scale:
             components.append('sigma2=%s' % self.sigma2)
         return 'SARIMAXParams(%s)' % ', '.join(components)
+
+    def _set_param_array(self, param_type, value, k_params, allow_infnan=False,
```

Comment:
Factored out this helper function from original param setters to handle setting partial param array with the `bool_indexer` arg. All free or full param setters use this help function except for `.params` and `.free_params`.